src/components/reusable/datePicker/datepicker.stories.js

```diff
@@ -1,114 +1,217 @@
-      table: {
-        defaultValue: { summary: DATE_PICKER_TYPES.SINGLE },
-      },
-    },
-    size: {
-      options: ['sm', 'md', 'lg'],
+    locale: { control: { type: 'text' } },
+    dateFormat: {
+      options: [
+        'Y-m-d',
+        'm-d-Y',
+        'd-m-Y',
+        'Y-m-d H:i',
+        'Y-m-d H:i:s',
+        'm-d-Y H:i:s',
+        'd-m-Y H:i:s',
+      ],
       control: { type: 'select' },
     },
-    minDate: {
-      control: { type: 'text' },
-    },
-    maxDate: {
-      control: { type: 'text' },
-    },
-    step: {
-      control: { type: 'text' },
+    mode: {
+      options: ['single', 'multiple'],
+      control: { type: 'select' },
     },
+    minDate: { control: { type: 'text' } },
+    maxDate: { control: { type: 'text' } },
   },
 };
 
-const args = {
-  unnamed: 'Date',
-  size: 'md',
-  name: 'datepicker',
-  value: '',
-  datePickerType: 'single',
-  caption: '',
+const disconnectFlatpickr = () => {
+  const calendarElements = document.querySelectorAll('.flatpickr-calendar');
+  calendarElements.forEach((calendar) => calendar.remove());
+};
+
+const InputTemplate = (args) => {
+  // prevents flatpickr calendar overlay from persisting on view change
+  useEffect(() => {
+    return () => {
+      disconnectFlatpickr();
+    };
+  }, []);
+
+  const anchorId =
+    args.nameAttr ||
+    `date-picker-button-${Math.random().toString(36).slice(2, 11)}`;
+
+  const placeholder = getPlaceholder(args.dateFormat);
+
+  return html`<kyn-date-picker
+    .nameAttr="${args.nameAttr}"
+    .locale="${args.locale}"
+    .dateFormat="${args.dateFormat}"
+    ?required="${args.required}"
+    .value="${args.value}"
+    .warnText="${args.warnText}"
+    .invalidText="${args.invalidText}"
+    .altFormat=${args.altFormat}
+    .disable="${args.disable}"
+    .enable="${args.enable}"
+    .mode="${args.mode}"
+    .caption="${args.caption}"
+    ?datePickerDisabled="${args.datePickerDisabled}"
+    ?twentyFourHourFormat="${args.twentyFourHourFormat}"
+    .minDate="${args.minDate}"
+    .maxDate="${args.maxDate}"
+    @on-change=${(e) => action(e.type)(e)}
+  >
+    <label
+      slot="label"
+      class="label-text"
+      for=${anchorId}
+      ?disabled=${args.datePickerDisabled}
+    >
+      ${args.required
+        ? html`<abbr
+            class="required"
+            title=${args._textStrings?.requiredText || 'Required'}
+            aria-label=${args._textStrings?.requiredText || 'Required'}
+            >*</abbr
+          >`
+        : null}
+      ${args.unnamed}
+    </label>
+    <input
+      slot="anchor"
+      type="text"
+      id=${anchorId}
+      name=${args.nameAttr}
+      placeholder=${placeholder}
+      ?disabled=${args.datePickerDisabled}
+      ?required=${args.required}
+      aria-invalid=${args._isInvalid ? 'true' : 'false'}
+    />
+    <span slot="icon" style="line-height: 1;">${unsafeSVG(overflowIcon)}</span>
```

Comment:
Should this be a calendar icon, why overflow?

Reply:
oh, yes this should be a calendar icon

Reply:
fixed this in migrating inputs from slot to internal